addon/components/img-lazy.js

```diff
@@ -50,34 +58,42 @@ export default Component.extend({
   alt: null,
   width: null,
   height: null,
+  srcset: null,
+  sizes: null,
 
   attributeBindings: [
     '_src:src',
+    '_srcset:srcset',
     'title',
     'alt',
     'width',
     'height',
+    'sizes'
   ],
 
   didInsertElement() {
-    get(this, 'observer').observe(this)
+    this._super(...arguments)
+    get(this, 'observer').observe(this.element)
   },
 
   willDestroyElement() {
-    get(this, 'observer').unobserve(this)
+    this._super(...arguments)
+    get(this, 'observer').unobserve(this.element)
   },
 
   loadImage() {
-    const src = get(this, 'src')
+    let src = get(this, 'src')
+    let srcset = get(this, 'srcset')
 
-    if (src === null) {
+    if (src === null && srcset === null) {
       return
     }
 
     this._setIsLoading()
```

Comment:
I'm not sure what the proper behavior would be to set/unset loading state when using `srcset` as we are waiting for the image elements loading event, maybe drop it completely once `srcset` is used? I'm also not familiar with how or if the onload event works with `srcset`. Do browsers supporting `srcset` even try to load the references `src` image if a `srcset` image matches?

Reply:
No, they wont try to load src if srcset is present. The presence of src attribute acts a fallback mechanism especially for those browsers which don't understand srcset attr.

I will need to look at onload behavior for srcset. So far it works in my PC.